File: sdk-rust/tests/openai_chat_test.rs
use crate::{common::cases::RunTestCaseOptions, test_group, test_set};
use llm_sdk::openai::*;
use std::{env, error::Error, sync::OnceLock};
use tokio::test;

fn openai_api_key() -> &'static String {
    static KEY: OnceLock<String> = OnceLock::new();

    KEY.get_or_init(|| {
        dotenvy::dotenv().ok();
        env::var("OPENAI_API_KEY").expect("OPENAI_API_KEY must be set")
    })
}

fn openai_model() -> OpenAIChatModel {
    OpenAIChatModel::new(
        "gpt-5.6-terra".to_string(),
        OpenAIChatModelOptions {
            api_key: openai_api_key().clone(),
            ..Default::default()
        },
    )
}

fn openai_audio_model() -> OpenAIChatModel {
    OpenAIChatModel::new(
        "gpt-audio-1.5".to_string(),
        OpenAIChatModelOptions {
            api_key: openai_api_key().clone(),
            ..Default::default()
        },
    )
}

fn no_reasoning_options() -> RunTestCaseOptions {
    RunTestCaseOptions {
        profile: Some("reasoning_disabled"),
    }
}

test_group!(
    openai_model(),
    text_generation,
    Some(no_reasoning_options())
);
test_group!(openai_model(), conversation, Some(no_reasoning_options()));
test_group!(openai_model(), tool_use, Some(no_reasoning_options()));
test_group!(
    openai_model(),
    structured_output,
    Some(no_reasoning_options())
);
test_group!(
    openai_model(),
    generation_options,
    Some(no_reasoning_options())
);
test_group!(openai_model(), source_input, Some(no_reasoning_options()));
test_group!(openai_model(), image_input);

test_set!(
    openai_audio_model(),
    generate_audio,
    Some(RunTestCaseOptions {
        profile: Some("openai_audio_mp3"),
    })
);

test_set!(
    openai_audio_model(),
    stream_audio,
    Some(RunTestCaseOptions {
        profile: Some("openai_audio_linear16"),
    })
);

File: sdk-js/src/openai/openai-chat.test.ts
import {
  runTestCase,
  runTestGroup,
  SHARED_BEHAVIOR_TEST_GROUPS,
  TEST_CASE_NAMES,
} from "#test-common/cases";
import assert from "node:assert";
import test, { suite } from "node:test";
import { OpenAIChatModel } from "./openai-chat.ts";

suite("OpenAIChatModel", () => {
  assert(process.env["OPENAI_API_KEY"], "OPENAI_API_KEY must be set");
  const model = new OpenAIChatModel({
    apiKey: process.env["OPENAI_API_KEY"],
    modelId: "gpt-5.6-terra",
  });

  const audioModel = new OpenAIChatModel({
    modelId: "gpt-audio-1.5",
    apiKey: process.env["OPENAI_API_KEY"],
  });

  const noReasoningOptions = { profile: "reasoning_disabled" };

  for (const group of SHARED_BEHAVIOR_TEST_GROUPS) {
    test(group, { timeout: 120 * 1000 }, (t) => {
      return runTestGroup(t, model, group, noReasoningOptions);
    });
  }

  test("image_input", (t) => runTestGroup(t, model, "image_input"));
  test(TEST_CASE_NAMES.GENERATE_AUDIO, (t) =>
    runTestCase(t, audioModel, TEST_CASE_NAMES.GENERATE_AUDIO, {
      profile: "openai_audio_mp3",
    }),
  );
  test(TEST_CASE_NAMES.STREAM_AUDIO, (t) =>
    runTestCase(t, audioModel, TEST_CASE_NAMES.STREAM_AUDIO, {
      profile: "openai_audio_linear16",
    }),
  );
});
